### PR TITLE
Updating the existing docker inventory script.

### PR DESCRIPTION
**Pull request by chouseknecht**
##### ISSUE TYPE
- New Module Pull Request
##### ANSIBLE VERSION

```
ansible 2.1.0 (docker_testing 9bb069f873) last updated 2016/04/18 12:36:03 (GMT -400)
  lib/ansible/modules/core: (detached HEAD cf01087a30) last updated 2016/04/05 16:31:53 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 204b4bab56) last updated 2016/04/05 16:31:56 (GMT -400)
  config file =
  configured module search path = Default w/o overrides
```
##### SUMMARY

Updated inventory script for docker.
- Added full API connection support include TLS and TLS_VERIFY options. Now works out of the box with docker tools. 
- Added defaults within the script so that it works without needing anything defined in docker.yml
- Removed hard-coded hostvar keys. docker_\* keys are created dynamically based on container inspection results. 
- Added image_<image name> grouping
- Added support for --host
- Added full set of command line args
- Improved documentation.

### CODE REVIEW

**Comment by chouseknecht** [comment]
@bcoca

**Comment by bcoca** [comment]
you got submodule ref updates in the commits

File: contrib/inventory/docker.py [hunk 1]
```diff
@@ -149,211 +366,502 @@
...
+            # use default definition
+            docker_host = def_host or self._args.docker_host or self._env_args.docker_host or DEFAULT_DOCKER_HOST
+            api_version = def_version or self._args.api_version or self._env_args.api_version or \
+                DEFAULT_DOCKER_API_VERSION
+            tls_hostname = def_tls_hostname or self._args.tls_hostname or self._env_args.tls_hostname
+            tls_verify = def_tls_verify or self._args.tls_verify or self._env_args.tls_verify or DEFAULT_TLS_VERIFY
+            tls = def_tls or self._args.tls or self._env_args.tls or DEFAULT_TLS
+            ssl_version = def_ssl_version or self._args.ssl_version or self._env_args.ssl_version
+            cert_path = def_cert_path or self._args.cert_path or self._env_args.cert_path
+            if cert_path == self._env_args.cert_path:
```
Comment (bcoca) [inline contrib/inventory/docker.py]: assumes cert_path is always set, should be:

`if cert_path and cert_path == self._env_args.cert_path:`

same for cacert_path and key_path

**Comment by bcoca** [comment]
fix both issues above and its good to go

**Comment by chouseknecht** [comment]
@bcoca Another good catch!  Thanks.
- Fixed the cert_path, key and cacert_path in get_hosts().
- And removed the submodules.

File: contrib/inventory/docker.py [hunk 2]
```diff
@@ -149,211 +366,507 @@
...
+            tls_verify = def_tls_verify or self._args.tls_verify or self._env_args.tls_verify or DEFAULT_TLS_VERIFY
+            tls = def_tls or self._args.tls or self._env_args.tls or DEFAULT_TLS
+            ssl_version = def_ssl_version or self._args.ssl_version or self._env_args.ssl_version
+
+            cert_path = def_cert_path or self._args.cert_path or self._env_args.cert_path
+            if cert_path and cert_path == self._env_args.cert_path:
+                    cert_path = os.path.join(cert_path, 'cert.pem')
+
+            cacert_path = def_cacert_path or self._args.cacert_path or self._env_args.cert_path
+            if cacertpath and cacert_path == self._env_args.cert_path:
```
Comment (bcoca) [inline contrib/inventory/docker.py:740]: cacertpath should be cacert_path

**Pull request merged**